tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/error/ErrorTest.java

```diff
@@ -137,7 +137,7 @@ public void customErrorDetailsTest() {
         BValue[] returns = BRunUtil.invoke(errorTestResult, "testCustomErrorDetails");
         Assert.assertEquals(returns[0].stringValue(), "trxErr {message:\"\", data:\"test\"}");
         Assert.assertEquals(((BError) returns[0]).getDetails().getType().getTag(), TypeTags.RECORD_TYPE_TAG);
-        Assert.assertEquals(((BError) returns[0]).getDetails().getType().getName(), "TrxErrorData & readonly");
+        Assert.assertEquals(((BError) returns[0]).getDetails().getType().getName(), "(TrxErrorData & readonly)");
```

Comment:
Is this type name change expected?

Reply:
Yes. In previous implementation, the type of the record returned by `detail()` function is not readonly, but in runtime, when the record is created, typeName is generated. That typeName did not have enclosing parenthesis. But now detail() returns a `readonly & DetailType`. So in CodeGen when the recordType is created, the full record name is taken from the tsymbol's name. So it has enclosing parenthesis.